Makefile

```diff
@@ -137,11 +149,48 @@ codegen:
 	controller-gen \
 		object:headerFile=hack/boilerplate.go.txt \
 		paths=./...
-	pnpm --dir=ui install --dev
-	pnpm --dir=ui run generate:schema
+
+.PHONY: codegen-docs
+codegen-docs:
 	npm install -g @bitnami/readme-generator-for-helm
 	bash hack/helm-docs/helm-docs.sh
 
+.PHONY: codegen-proto
+codegen-proto:
+	# Vendor dependencies for protobuf code generation
+	go mod tidy
+	go mod vendor
+
+	# Generate protobuf code from Kubebuilder structs
+	GOPATH=${GOPATH} go-to-protobuf \
+		--go-header-file=./hack/boilerplate.go.txt \
+		--packages=github.com/akuity/kargo/api/v1alpha1 \
+		--apimachinery-packages=${APIMACHINERY_PKGS} \
+		--proto-import $(CURDIR)/vendor
+	# Copy generated code to the working directory
+	cp -R ${GOPATH}/src/github.com/akuity/kargo/api $(CURDIR)
+
+	# Generate protobuf code (Go)
+	buf generate api
+	# Inject generated protobuf tag to Kubebuilder structs
+	go run $(CURDIR)/hack/codegen/prototag/main.go \
+		-src-dir=$(CURDIR)/pkg/api/v1alpha1 \
+		-dst-dir=$(CURDIR)/api/v1alpha1
+	# Clean up intermediate file
+	rm -r $(CURDIR)/pkg/api/v1alpha1
```

Comment:
@krancour As far as I know, there is no same concept with Go's defer. Could you recommend any way to clean up intermediate files even though the previous commands are failed?

Reply:
TIL

Reply:
I just found that `.DELETE_ON_ERROR` doesn't work against directories... https://savannah.gnu.org/bugs/?16372

I resolved it by using bash `trap`: https://github.com/akuity/kargo/pull/1515/commits/c1135af5e095657f5720a434f9ab60725282519d